src/Common/src/CoreLib/System/Collections/Generic/ValueListBuilder.cs

```diff
@@ -13,12 +13,14 @@ namespace System.Collections.Generic
         private Span<T> _span;
         private T[] _arrayFromPool;
         private int _pos;
+        private readonly bool _clearArray;
```

Comment:
We should not be making ValueListBuilder more expensive by adding cruft like this to it.

Reply:
As I'm not using these changes here I will revert them.

Reply:
I already reverted the change but I wouldn't call that addition cruft. In cases where you want to use ValueListBuilder with types that need to cleaned up, the clearArray overload is handy. In my case I didn't use it as it is faster to partially clean it up instead of using Array.Clear.

Reply:
Pint is it's not pay for play though, it makes everyone use more stack

Reply:
You cannot really build common helper classes like these to support everything that one might potentially want to do. It kills their value. They need to stay focused on the mainstream use.